Hamilton_Matthew/history.py

```diff
@@ -0,0 +1,173 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        #request.session stuff
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.        
+                    
+        if not request.user.is_authenticated:
+            request.session['new_prod'] = -1
+
+        new_prod_id = int(request.session['new_prod'])
+        print ("new_prod_id: ")
+        print(new_prod_id)
+        
+        if new_prod_id > 0:
+            print("woah")
+
+            if new_prod_id in request.session['last_five']:
+                print("the item is already in the list, removing: ")
+                print(new_prod_id)
+                request.session['last_five'].remove(new_prod_id)
+            
+            print("adding: ")
+            print(new_prod_id)
+            request.session['last_five'].insert(1, new_prod_id)
+
+        request.last_five = []
+
+        print(request.session['last_five'])
+
+
+        for i in request.session['last_five']:
+            print("i: " + str(i))
+            request.last_five.append(cmod.Product.objects.get(id = i))
+
+        
+
+        response = self.get_response(request)
+
+        request.session['last_five'] = []
+
+        for product in request.last_five:
+            request.session['last_five'].append(product.id)
+
+        while len(request.session['last_five']) > 7:
+            request.session['last_five'].pop()
+
+        return response
+
+
+
+
```

Comment:
The comments below can be deleted or integrated into the code above.

Reply:
I shall remove them